In [ ]:
##To load the file when we are using Google Colab
##from google.colab import drive
##drive.mount('/content/drive')

In [6]:
# Libraries to help with reading and manipulating data
import numpy as np
import pandas as pd

# Libraries to help with data visualization
import matplotlib.pyplot as plt
import seaborn as sns

# Command to tell Python to actually display the graphs
%matplotlib.inline


UsageError: Line magic function `%matplotlib.inline` not found.


In [3]:
import IPython
print(IPython.sys_info())

{'commit_hash': 'add5877a4',
 'commit_source': 'installation',
 'default_encoding': 'cp1252',
 'ipython_path': 'C:\\Users\\Sudhanshu\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython',
 'ipython_version': '8.8.0',
 'os_name': 'nt',
 'platform': 'Windows-10-10.0.22621-SP0',
 'sys_executable': 'C:\\ProgramData\\Anaconda3\\python.exe',
 'sys_platform': 'win32',
 'sys_version': '3.9.13 (main, Aug 25 2022, 23:51:50) [MSC v.1916 64 bit '
                '(AMD64)]'}


In [5]:
pip --version

pip 22.2.2 from C:\ProgramData\Anaconda3\lib\site-packages\pip (python 3.9)

Note: you may need to restart the kernel to use updated packages.


In [7]:
pip list

Package                       Version
----------------------------- --------------------
alabaster                     0.7.12
anaconda-client               1.11.0
anaconda-navigator            2.3.2
anaconda-project              0.11.1
anyio                         3.5.0
appdirs                       1.4.4
argon2-cffi                   21.3.0
argon2-cffi-bindings          21.2.0
arrow                         1.2.2
astroid                       2.11.7
astropy                       5.1
atomicwrites                  1.4.0
attrs                         21.4.0
Automat                       20.2.0
autopep8                      1.6.0
Babel                         2.9.1
backcall                      0.2.0
backports.functools-lru-cache 1.6.4
backports.tempfile            1.0
backports.weakref             1.0.post1
bcrypt                        3.2.0
beautifulsoup4                4.11.1
binaryornot                   0.4.4
bitarray                      2.5.1
bkcharts                      0.2
blac

In [8]:
pip install --upgrade ipython

Defaulting to user installation because normal site-packages is not writeable
     -------------------------------------- 775.8/775.8 kB 7.0 MB/s eta 0:00:00
  Using cached stack_data-0.6.2-py3-none-any.whl (24 kB)
  Using cached asttokens-2.2.1-py2.py3-none-any.whl (26 kB)
  Using cached pure_eval-0.2.2-py3-none-any.whl (11 kB)
  Using cached executing-1.2.0-py2.py3-none-any.whl (24 kB)
Note: you may need to restart the kernel to use updated packages.


  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
spyder 5.2.2 requires pyqt5<5.13, which is not installed.
spyder 5.2.2 requires pyqtwebengine<5.13, which is not installed.
spyder 5.2.2 requires ipython<8.0.0,>=7.6.0, but you have ipython 8.8.0 which is incompatible.
spyder-kernels 2.2.1 requires ipython<8,>=7.6.0; python_version >= "3", but you have ipython 8.8.0 which is incompatible.


In [4]:
print(IPython.sys_info())

{'commit_hash': 'add5877a4',
 'commit_source': 'installation',
 'default_encoding': 'cp1252',
 'ipython_path': 'C:\\Users\\Sudhanshu\\AppData\\Roaming\\Python\\Python39\\site-packages\\IPython',
 'ipython_version': '8.8.0',
 'os_name': 'nt',
 'platform': 'Windows-10-10.0.22621-SP0',
 'sys_executable': 'C:\\ProgramData\\Anaconda3\\python.exe',
 'sys_platform': 'win32',
 'sys_version': '3.9.13 (main, Aug 25 2022, 23:51:50) [MSC v.1916 64 bit '
                '(AMD64)]'}


In [11]:
df = pd.read_csv("Automobile.csv")

In [12]:
df.head()

symboling  normalized_losses         make fuel_type aspiration  \
0          3                168  alfa-romero       gas        std   
1          3                168  alfa-romero       gas        std   
2          1                168  alfa-romero       gas        std   
3          2                164         audi       gas        std   
4          2                164         audi       gas        std   

  number_of_doors   body_style drive_wheels engine_location  wheel_base  ...  \
0             two  convertible          rwd           front        88.6  ...   
1             two  convertible          rwd           front        88.6  ...   
2             two    hatchback          rwd           front        94.5  ...   
3            four        sedan          fwd           front        99.8  ...   
4            four        sedan          4wd           front        99.4  ...   

   engine_size  fuel_system  bore  stroke compression_ratio horsepower  \
0          130         mpfi  3.47    2.68               9.0        111   
1          130         mpfi  3.47    2.68               9.0        111   
2          152         mpfi  2.68    3.47               9.0        154   
3          109         mpfi  3.19    3.40              10.0        102   
4          136         mpfi  3.19    3.40               8.0        115   

   peak_rpm city_mpg  highway_mpg  price  
0      5000       21           27  13495  
1      5000       21           27  16500  
2      5000       19           26  16500  
3      5500       24           30  13950  
4      5500       18           22  17450  

[5 rows x 26 columns]

In [13]:
df.shape

(201, 26)

In [14]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 201 entries, 0 to 200
Data columns (total 26 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   symboling            201 non-null    int64  
 1   normalized_losses    201 non-null    int64  
 2   make                 201 non-null    object 
 3   fuel_type            201 non-null    object 
 4   aspiration           201 non-null    object 
 5   number_of_doors      201 non-null    object 
 6   body_style           201 non-null    object 
 7   drive_wheels         201 non-null    object 
 8   engine_location      201 non-null    object 
 9   wheel_base           201 non-null    float64
 10  length               201 non-null    float64
 11  width                201 non-null    float64
 12  height               201 non-null    float64
 13  curb_weight          201 non-null    int64  
 14  engine_type          201 non-null    object 
 15  number_of_cylinders  201 non-null    obj

- There are attributes of different types (*int*, *float*, *object*) in the data.

In [16]:
df.describe().T

count          mean          std      min      25%  \
symboling          201.0      0.840796     1.254802    -2.00     0.00   
normalized_losses  201.0    125.189055    33.572966    65.00   101.00   
wheel_base         201.0     98.797015     6.066366    86.60    94.50   
length             201.0    174.200995    12.322175   141.10   166.80   
width              201.0     65.889055     2.101471    60.30    64.10   
height             201.0     53.766667     2.447822    47.80    52.00   
curb_weight        201.0   2555.666667   517.296727  1488.00  2169.00   
engine_size        201.0    126.875622    41.546834    61.00    98.00   
bore               201.0      3.329701     0.268166     2.54     3.15   
stroke             201.0      3.261741     0.317875     2.07     3.11   
compression_ratio  201.0     10.164279     4.004965     7.00     8.60   
horsepower         201.0    103.263682    37.389372    48.00    70.00   
peak_rpm           201.0   5121.393035   479.624905  4150.00  4800.00   
city_mpg           201.0     25.179104     6.423220    13.00    19.00   
highway_mpg        201.0     30.686567     6.815150    16.00    25.00   
price              201.0  13207.129353  7947.066342  5118.00  7775.00   

                        50%       75%       max  
symboling              1.00      2.00      3.00  
normalized_losses    122.00    150.00    256.00  
wheel_base            97.00    102.40    120.90  
length               173.20    183.50    208.10  
width                 65.50     66.60     72.00  
height                54.10     55.50     59.80  
curb_weight         2414.00   2926.00   4066.00  
engine_size          120.00    141.00    326.00  
bore                   3.31      3.58      3.94  
stroke                 3.29      3.46      4.17  
compression_ratio      9.00      9.40     23.00  
horsepower            95.00    116.00    262.00  
peak_rpm            5200.00   5500.00   6600.00  
city_mpg              24.00     30.00     49.00  
highway_mpg           30.00     34.00     54.00  
price              10295.00  16500.00  45400.00

In [17]:
df.describe(include='all').T

count unique     top freq          mean          std  \
symboling            201.0    NaN     NaN  NaN      0.840796     1.254802   
normalized_losses    201.0    NaN     NaN  NaN    125.189055    33.572966   
make                   201     22  toyota   32           NaN          NaN   
fuel_type              201      2     gas  181           NaN          NaN   
aspiration             201      2     std  165           NaN          NaN   
number_of_doors        201      2    four  114           NaN          NaN   
body_style             201      5   sedan   94           NaN          NaN   
drive_wheels           201      3     fwd  118           NaN          NaN   
engine_location        201      2   front  198           NaN          NaN   
wheel_base           201.0    NaN     NaN  NaN     98.797015     6.066366   
length               201.0    NaN     NaN  NaN    174.200995    12.322175   
width                201.0    NaN     NaN  NaN     65.889055     2.101471   
height               201.0    NaN     NaN  NaN     53.766667     2.447822   
curb_weight          201.0    NaN     NaN  NaN   2555.666667   517.296727   
engine_type            201      6     ohc  145           NaN          NaN   
number_of_cylinders    201      7    four  157           NaN          NaN   
engine_size          201.0    NaN     NaN  NaN    126.875622    41.546834   
fuel_system            201      8    mpfi   92           NaN          NaN   
bore                 201.0    NaN     NaN  NaN      3.329701     0.268166   
stroke               201.0    NaN     NaN  NaN      3.261741     0.317875   
compression_ratio    201.0    NaN     NaN  NaN     10.164279     4.004965   
horsepower           201.0    NaN     NaN  NaN    103.263682    37.389372   
peak_rpm             201.0    NaN     NaN  NaN   5121.393035   479.624905   
city_mpg             201.0    NaN     NaN  NaN     25.179104      6.42322   
highway_mpg          201.0    NaN     NaN  NaN     30.686567      6.81515   
price                201.0    NaN     NaN  NaN  13207.129353  7947.066342   

                        min     25%      50%      75%      max  
symboling              -2.0     0.0      1.0      2.0      3.0  
normalized_losses      65.0   101.0    122.0    150.0    256.0  
make                    NaN     NaN      NaN      NaN      NaN  
fuel_type               NaN     NaN      NaN      NaN      NaN  
aspiration              NaN     NaN      NaN      NaN      NaN  
number_of_doors         NaN     NaN      NaN      NaN      NaN  
body_style              NaN     NaN      NaN      NaN      NaN  
drive_wheels            NaN     NaN      NaN      NaN      NaN  
engine_location         NaN     NaN      NaN      NaN      NaN  
wheel_base             86.6    94.5     97.0    102.4    120.9  
length                141.1   166.8    173.2    183.5    208.1  
width                  60.3    64.1     65.5     66.6     72.0  
height                 47.8    52.0     54.1     55.5     59.8  
curb_weight          1488.0  2169.0   2414.0   2926.0   4066.0  
engine_type             NaN     NaN      NaN      NaN      NaN  
number_of_cylinders     NaN     NaN      NaN      NaN      NaN  
engine_size            61.0    98.0    120.0    141.0    326.0  
fuel_system             NaN     NaN      NaN      NaN      NaN  
bore                   2.54    3.15     3.31     3.58     3.94  
stroke                 2.07    3.11     3.29     3.46     4.17  
compression_ratio       7.0     8.6      9.0      9.4     23.0  
horsepower             48.0    70.0     95.0    116.0    262.0  
peak_rpm             4150.0  4800.0   5200.0   5500.0   6600.0  
city_mpg               13.0    19.0     24.0     30.0     49.0  
highway_mpg            16.0    25.0     30.0     34.0     54.0  
price                5118.0  7775.0  10295.0  16500.0  45400.0

- The car price ranges from 5118 to 45400 units.
- The car weight ranges from 1488 to 4066 units.
- The most common car make in the data is of Toyota.